hypothesis-python/src/hypothesis/strategies/_internal/misc.py

```diff
@@ -79,28 +88,30 @@ def none() -> SearchStrategy[None]:
     return just(None)
 
 
-class Nothing(SearchStrategy):
-    def calc_is_empty(self, recur):
+class Nothing(SearchStrategy["Never"]):
+    def calc_is_empty(self, recur: RecurT) -> bool:
         return True
 
-    def do_draw(self, data):
+    def do_draw(self, data: ConjectureData) -> NoReturn:
         # This method should never be called because draw() will mark the
         # data as invalid immediately because is_empty is True.
         raise NotImplementedError("This should never happen")
 
-    def calc_has_reusable_values(self, recur):
+    def calc_has_reusable_values(self, recur: RecurT) -> bool:
         return True
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return "nothing()"
 
-    def map(self, f):
+    def map(self, pack: Callable[[Ex], T]) -> SearchStrategy["Never"]:
         return self
 
-    def filter(self, f):
+    def filter(self, condition: Callable[[Ex], Any]) -> "SearchStrategy[Never]":
         return self
 
-    def flatmap(self, f):
+    def flatmap(
+        self, expand: Callable[[Ex], "SearchStrategy[T]"]
+    ) -> "SearchStrategy[Never]":
```

Comment:
I think that, because there are no typevars in the return type, we should replace those in the params with `Any`.

iirc this is the typecheckers' intrepretation of our current code, but explicit is better than implicit and... it really doesn't matter what your callable takes and returns, because it'll never be invoked anyway.